frontend/src/hooks/useKeycloakWrapper.tsx

```diff
@@ -1,60 +1,140 @@
 import { useKeycloak } from '@react-keycloak/web';
-import { SYSTEM_ADMINISTRATOR, AGENCY_ADMINISTRATOR } from 'constants/strings';
+import { Roles } from 'constants/roles';
 
+/**
+ * IUserInfo interface, represents the userinfo provided by keycloak.
+ */
 export interface IUserInfo {
+  displayName?: string;
   username: string;
   name?: string;
   preferred_username?: string;
   firstName?: string;
   lastName?: string;
   email: string;
-  Groups: any[];
+  groups: string[];
+  roles: string[];
   given_name?: string;
   family_name?: string;
   agencies: number[];
 }
 
-function useKeycloakWrapper() {
+/**
+ * IKeycloak interface, represents the keycloak object for the authenticated user.
+ */
+interface IKeycloak {
+  obj: any;
+  displayName?: string;
+  username: string;
+  name?: string;
+  preferred_username?: string;
+  firstName?: string;
+  lastName?: string;
+  email?: string;
+  roles: string[];
+  agencyId?: number;
+  isAdmin: boolean;
+  hasRole(role?: string | Array<string>): boolean;
+  hasClaim(claim?: string | Array<string>): boolean;
+  hasAgency(agency?: number): boolean;
+}
+
+/**
+ * Provides extension methods to interact with the `keycloak` object.
+ */
+function useKeycloakWrapper(): IKeycloak {
   const { keycloak } = useKeycloak();
-  const hasClaim = (claim?: string) => {
-    return !claim || keycloak?.hasResourceRole(claim);
-  };
   const userInfo = keycloak?.userInfo as IUserInfo;
-  const hasRole = (role?: string) => {
-    return !role || userInfo?.Groups?.includes(role);
+
+  /**
+   * Determine if the user has the specified 'claim'
+   * @param claim - The name of the claim
+   */
+  const hasClaim = (claim?: string | Array<string>): boolean => {
+    return (
+      !claim ||
+      (typeof claim === 'string'
+        ? userInfo?.roles?.includes(claim)
+        : claim.some(c => userInfo?.roles?.includes(c)))
+    );
```

Comment:
The code above that has to check string or array is exactly what I alluded earlier. In my opinion always checking for the array version would lead to less bugs in the future. Also, less confusing to people learning the claims system in the front-end

Reply:
I have no issues if we want to enforce `[ "role" ]` if it's truly simpler for use.  I just don't think it is simpler for use.  The complexity is already resolved within the function and now provides easier use of the function everywhere else.

Reply:
It seems we have differing opinions which is okay. You are the author of the code so I'm fine with it. We can always revisit later if it causes issues

Reply:
All I'm saying is we should hear the team on this.  I'm of the opinion that this is beneficial for use, your are saying it increases complexity and therefore increases future bug opportunities.